### DISCUSSION TITLE
static IP (not for the config portal)

### DISCUSSION THREAD

**Discussion by stanelie** [khoih-prog/ESP_WiFiManager_Lite#22] [Q&A] [answered]
Hello.

Is it possible to use this while setting a static IP address within the code?

I would like the config portal to not show the option to set a static IP address while using the one that is configured within the code from a formula that derives it from the mac address and a special number  (I am using this to build a Artnet device, the Artnet manufacturer requires this).

Thanks!

**Comment by khoih-prog** [comment, marked as answer]
To use static IP, you have to use the following function in your code

https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/2244ae0e98c1778a53283364c13d5872cec9e34c/src/ESP_WiFiManager_Lite.h#L1020-L1040

before

https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/2244ae0e98c1778a53283364c13d5872cec9e34c/examples/ESP_WiFi/ESP_WiFi.ino#L106


Certainly, to have those static IPs (staIP, GW, SN, DNS1, DNS2, etc) , you either 

- use Config Portal to get it by adding dynamicParam in [**dynamicParams.h**](https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/main/examples/ESP_WiFi/dynamicParams.h)
- get from LittleFS/SPIFFS file by writing code for this
- many more other options you can think of

**Comment by stanelie** [comment]
Thanks again!
